src/telemetry/telemetry.c

```diff
@@ -60,27 +73,19 @@ telemetry_parse_version(const char *json, VersionInfo *installed_version, Versio
 
 	result->versionstr = text_to_cstring(DatumGetTextPP(version));
 
-	result->is_up_to_date = false;
-
 	if (result->versionstr == NULL)
 	{
 		result->errhint = "no version string in response";
 		return false;
 	}
 
-	/*
-	 * Now parse the version string. We expect format to be
-	 * XX.XX.XX-<prerelease_tag>, and if not, we error out
-	 */
-	if (!version_parse(result->versionstr, &result->vinfo))
+	for (i = 0; i < NUM_VERSION_CHARS_TO_PRINT && i < strlen(result->versionstr); i++)
 	{
-		result->errhint = psprintf("parsing failed for version string \"%s\"", result->versionstr);
-		return false;
+		if (result->versionstr[i] == '\0' || (!isalpha(result->versionstr[i]) && !isdigit(result->versionstr[i]) && result->versionstr[i] != '.'))
+			break;
 	}
 
-	if (version_cmp(installed_version, &result->vinfo) >= 0)
-		result->is_up_to_date = true;
-
+	result->num_printable_chars = i;
```

Comment:
shouldn't we just `\0' terminate the string here instead of saving the num printable chars?

In other words:
```
result->versionstr[i] = '\0'
```

But I am not sure it is right to modify or limit the output of the version string at all. Seems to me that either we just print what the server tells us or any failure to validate the entire string is a failure. I find it a bit strange to print just part of the string.